File: 백준/단계 별로 풀어보기/기본_수학_2/4153_직각삼각형.py
import sys
input = sys.stdin.readline
IP = lambda: map(int, input().split())

def solve():
    while True:
        a, b, c = sorted(IP())
        if not (a or b or c): break
        print("right" if a ** 2 + b ** 2 == c ** 2 else "wrong")
    
if "__main__" == __name__: solve()

File: 백준/단계 별로 풀어보기/기본_수학_2/4153_직각삼각형.cpp
#include <iostream>
#include <vector>
#include <string>
#include <algorithm>
#include <cmath>
using namespace std;

int main()
{
	cin.tie(NULL);
	cout.tie(NULL);
	ios_base::sync_with_stdio(false);
	
	int input, T;
	while (true){
	    vector<int> L;
    	T = 3;
    	while (T--){
    	    cin >> input;
    	    L.push_back(input);
    	}
    	if (L[0] == 0 && L[1] == 0 && L[2] == 0) break;
    	sort(L.begin(), L.end());
    	cout << ((L[0] * L[0] + L[1] * L[1] == L[2] * L[2]) ? ("right\n") : ("wrong\n"));
	}
}